src/AutorestSwift/View Models/OperationViewModel.swift

```diff
@@ -118,73 +91,54 @@ struct OperationParameters {
                     switch property {
                     case let .grouped(groupProperty):
                         for param in groupProperty.originalParameter where param.paramLocation == .path {
-                            signatureParameterViewModel
-                                .append(ParameterViewModel(from: param))
+                            signatureViewModel.append(ParameterViewModel(from: param))
                         }
                     default:
                         continue
                     }
                 }
-            } else if param.belongsInSignature() {
-                signatureParameterViewModel.append(ParameterViewModel(from: param))
+            } else if param.belongsInSignature(model: model) {
+                signatureViewModel.append(ParameterViewModel(from: param))
             }
         }
-        self.signature = signatureParameterViewModel
-
-        self.methodDecoding = methodDecoding
-        self.header = header
-        self.query = query
-        self.explodeQuery = explodeQuery
-        self.path = path
-        self.hasOptionalParams = !(query.optional.isEmpty && header.optional.isEmpty && explodeQuery.optional.isEmpty)
-    }
-}
-
-struct Params {
-    // Query Params/Header in initializer
-    var required: [KeyValueViewModel]
-    // Query Params/Header need to add Nil check
-    var optional: [KeyValueViewModel]
-    // Whether to 'var' or 'let' in generated code for the param declaration
-    var declaration: String = "var"
-
-    init() {
-        self.required = [KeyValueViewModel]()
-        self.optional = [KeyValueViewModel]()
-    }
-
-    var isEmpty: Bool {
-        return required.isEmpty && optional.isEmpty
+        self.all = params
+        self.signature = signatureViewModel
+        self.explode = explodeParams
+        self.declaration = explode.isEmpty ? "let" : "var"
     }
 }
 
 enum BodyParamStrategy: String {
     case plain
+    case plainNullable
     case flattened
     case unixTime
-    case plainNullable
     case byteArray
+    case base64ByteArray
     case constant
     case decimal
     case data
     case string
-    case date
-    case dateTime
-    case dateTimeRfc1123
     case time
 }
 
 struct BodyParams {
-    let param: ParameterViewModel
+    /// The `ParameterViewModel` corresponding to the body param
+    var param: ParameterViewModel
```

Comment:
can this bet 'let' instead of 'var'?

Reply:
No because OperationParam can update the name. This was to get rid of an initializer that had an optional `withName` in the signature.